### DISCUSSION TITLE
Why does Fastmate refresh when waking from sleep?

### DISCUSSION THREAD

**Discussion by sddgit** [joelekstrom/fastmate#43] [Q&A] [answered]
I've noticed that the Fastmate "page" refreshes every time I wake my MacBook Pro from sleep. A tab in Safari doesn't do this. Do you force this on purpose? I ask because I have been conversing with Fastmail support for some time regarding an issue where a Fastmail tab in Safari, after waking, will have a Safari message at the top saying the page has been refreshed because it is using significant resources. Fastmail thinks this is a problem with Safari because they have seen it with other web sites. So I'm wondering whether you're doing the refresh on purpose because you've seen Fastmail using a lot of CPU after a wake, and refreshing gets around the problem. Regardless of why you do it, I do believe it solves a problem!

**Comment by joelekstrom** [comment, marked as answer]
Hi! You’re correct, this is something I added to fix some slowness issues that sometimes happened after computer sleep. It may well be related to the issue that Safari flags since Fastmate uses the Safari engine.

I could add a setting for it if it’s something you’d like to turn off

**Reply by sddgit** [reply]
Ok, that is interesting! I’m more than happy with the way it is. Solves a problem really. Before discovering Fastmate, I tried some single-site-browsers, like Unite (also based on WebKit). Sometimes it just consumed CPU forever after a wake, because there was no mechanism to “trap” any excess resource utilisation, like the native Safari does. It’s really hard to say whether this is a problem with Safari/WebKit or the Fastmail implementation. Either way, I’m very happy with Fastmate!

**Reply by joelekstrom** [reply]
Great to hear! Thanks 😊
